pkg/pillar/cmd/baseosmgr/handledownload.go

```diff
@@ -479,3 +479,18 @@ func downloaderSubscription(ctx *baseOsMgrContext, objType string) *pubsub.Subsc
 	}
 	return sub
 }
+
+func allowNonFreePort(ctx *baseOsMgrContext, objType string) bool {
```

Comment:
Does it make sense to add case for app image download also and make this function available to zedmanager also.

Reply:
In principle, but since we have a baseOsMgrContext as argument that doesn't quite work. And the objType values are local to the agent.

Let me move this stuff into downloadertypes.go and make it be common.

Reply:
I've moved it to common code.